src/app/util/attribute-storage.cpp

```diff
@@ -480,15 +480,15 @@ static EmberAfStatus typeSensitiveMemCopy(ClusterId clusterId, uint8_t * dest, u
         {
             return EMBER_ZCL_STATUS_INSUFFICIENT_SPACE;
         }
-        emberAfCopyString(dest, src, static_cast<uint8_t>(bufferSize - 1));
+        emberAfCopyString(dest, src, static_cast<size_t>(bufferSize - 1));
```

Comment:
Why not just remove this cast?  If that leads to signed-vs-unsigned issues, change the type of bufferSize in this function to `size_t`?
```suggestion
        emberAfCopyString(dest, src, bufferSize - 1);
```